builtin/plugins/chainconfig/chainconfig.go

```diff
@@ -542,3 +558,78 @@ func removeFeature(ctx contract.Context, name string) error {
 }
 
 var Contract plugin.Contract = contract.MakePluginContract(&ChainConfig{})
+
+func (c *ChainConfig) SetValidatorInfo(ctx contract.Context, req *SetValidatorInfoRequest) error {
+	if !ctx.FeatureEnabled(loomchain.ChainCfgVersion1_2, false) {
+		return ErrFeatureNotEnabled
+	}
+
+	if req.BuildNumber == 0 {
+		return ErrInvalidRequest
+	}
+	senderAddr := ctx.Message().Sender
+	validators, err := getCurrentValidators(ctx)
+	if err != nil {
+		return err
+	}
+	isValidator := false
+	for _, validator := range validators {
+		if validator.Compare(senderAddr) == 0 {
+			isValidator = true
+			break
+		}
+	}
+	if !isValidator {
+		return ErrNotAuthorized
+	}
+	return setValidatorInfo(ctx, senderAddr, req.BuildNumber)
+}
+
+func setValidatorInfo(ctx contract.Context, addr loom.Address, buildNumber uint64) error {
+	validator := &ValidatorInfo{
+		Address:     addr.MarshalPB(),
+		BuildNumber: buildNumber,
+		UpdatedAt:   uint64(time.Now().Unix()),
+	}
+	return ctx.Set(validatorInfoKey(addr), validator)
+}
+
+func (c *ChainConfig) GetValidatorInfo(ctx contract.StaticContext, req *GetValidatorInfoRequest) (*GetValidatorInfoResponse, error) {
+	if req.Address == nil {
+		return nil, ErrInvalidRequest
+	}
+	address := loom.UnmarshalAddressPB(req.Address)
+	validatorInfo, err := getValidatorInfo(ctx, address)
+	if err != nil {
+		return nil, err
+	}
+	return &GetValidatorInfoResponse{
+		Validator: validatorInfo,
+	}, nil
+}
+
+func getValidatorInfo(ctx contract.StaticContext, addr loom.Address) (*ValidatorInfo, error) {
+	var validatorInfo ValidatorInfo
+	err := ctx.Get(validatorInfoKey(addr), &validatorInfo)
```

Comment:
It's better not to return an error if the validator info isn't found:
```golang
if err == contract.ErrNotFound {
  return &ValidatorInfo{}, nil
} else if err != nil {
  return nil, errors.Wrap(err, "failed to retrieve validator info")
}
```
Makes the client side code a bit less messy.